x-pack/plugins/embeddable_enhanced/public/drilldowns/embeddable_to_url_drilldown/embeddable_to_url_drilldown.tsx

```diff
@@ -0,0 +1,286 @@
+  implements DrilldownDefinition<UrlDrilldownConfig, ActionContext> {
+  public readonly id = 'EMB_TO_URL_DRILLDOWN';
+
+  public readonly minimalLicense = 'gold';
+
+  public readonly order = 8;
+
+  public readonly getDisplayName = () => 'Go to URL';
+
+  public readonly euiIcon = 'link';
+
+  constructor(private params: Params) {}
+
+  private readonly ReactCollectConfig: React.FC<CollectConfigProps> = ({
+    config,
+    onConfig,
+    context,
+  }) => {
+    const { getGlobalScope } = this.params;
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const scope = React.useMemo(
+      () => buildScope(getGlobalScope(), getContextScopeFromEmbeddable(context.embeddable)),
+      [getGlobalScope, context]
+    );
+
+    return <UrlDrilldownCollectConfig config={config} onConfig={onConfig} scope={scope} />;
+  };
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
+
+  public readonly createConfig = () => ({
+    url: '',
+    openInNewTab: false,
+  });
+
+  public readonly isConfigValid = (config: UrlDrilldownConfig): config is UrlDrilldownConfig => {
+    if (!config.url) return false;
+    return isValidUrl(config.url);
+  };
+
+  /**
+   * `getHref` is need to support mouse middle-click and Cmd + Click behavior
+   * to open a link in new tab.
+   */
+  public readonly getHref = async (config: UrlDrilldownConfig, context: ActionContext) => {
+    const globalScope = this.params.getGlobalScope();
+    const contextScope = getContextScopeFromEmbeddable(context.embeddable);
+    const eventScope = await getEventScopeFromActionContext(
+      context,
+      this.params.getDataActionsHelpers()
+    );
+
+    const scope = buildScope(globalScope, contextScope, eventScope);
+    const url = compile(config.url, scope);
+
+    return url;
+  };
+
+  public readonly execute = async (config: UrlDrilldownConfig, context: ActionContext) => {
+    const url = await this.getHref(config, context);
+
+    if (config.openInNewTab) {
+      window.open(url, '_blank', 'noopener');
+    } else {
+      window.location.href = url;
+    }
+  };
+}
+
+function getContextScopeFromEmbeddable(embeddable?: IEmbeddable): EmbeddableContextScope {
+  if (!embeddable) return {};
+  const input = embeddable.getInput();
+  const output = embeddable.getOutput();
+  // TODO: type it better
+  return {
+    panelId: input.id,
+    panelTitle: output.title,
+    ..._.pick(input, ['query', 'timeRange', 'filters']),
+    ...(output.savedObjectId
+      ? { savedObjectId: output.savedObjectId }
+      : _.pick(input, 'savedObjectId')),
+  };
+}
+
+async function getEventScopeFromActionContext(
+  context: ActionContext,
+  { createFiltersFromRangeSelectAction, createFiltersFromValueClickAction }: DataActionsHelpers
+): Promise<EmbeddableTriggerEventScope> {
+  const filtersFromEvent = await (async () => {
+    try {
+      if (isRangeSelectTriggerContext(context))
+        return await createFiltersFromRangeSelectAction(context.data);
```

Comment:
as mentioned in the doc, i would prefer if we would not postprocess context.data and expose it directly. it will give more power to the user and it will be easier to extend url drilldown with future triggers.